language/proto/resolve.go

```diff
@@ -34,7 +34,22 @@ func (_ *protoLang) Imports(c *config.Config, r *rule.Rule, f *rule.File) []reso
 	rel := f.Pkg
 	srcs := r.AttrStrings("srcs")
 	imports := make([]resolve.ImportSpec, len(srcs))
+	pc := GetProtoConfig(c)
 	for i, src := range srcs {
+		// If strip_import_prefix is a relative path (not starting with a slash), it's taken as a package-relative one.
```

Comment:
`rel` shouldn't be modified. Instead, determine the prefix to strip and the prefix to add outside the of the loop. Within the loop, try to strip the old prefix and add the new prefix.

If `rel` doesn't start with the prefix to strip, skip indexing these files. An error should be reported by the configurer when the prefix is set.